pkg/tracer/tracer_linux.go

```diff
@@ -610,9 +610,52 @@ func (t *LinuxTracer) dispatchPerfEvent(event *common.PerfEvent) {
 		}
 		t.tcpConnInsights[httpRes.Connection] = conn
 
-	} else if event.MySQLGreeting != nil {
-		logger.Tracef("mysql greeting: %v", event.MySQLGreeting)
-		mysqlGreeting := event.MySQLGreeting
+	} else if event.MySQLGreetingV4 != nil {
+		logger.Warnf("mysql greeting: %v", event.MySQLGreetingV4)
+		mysqlGreeting := event.MySQLGreetingV4
+
+		conn, ok := t.tcpConnInsights[mysqlGreeting.Connection]
+		if !ok {
+			conn = ConnInsight{
+				ApplicationProtocol: "mysql",
+				HttpMetrics:         make(map[HttpCode]*ddsketch.DDSketch),
+			}
+		}
+		conn.ApplicationProtocol = "mysql"
+		t.tcpConnInsights[mysqlGreeting.Connection] = conn
+	} else if event.HTTPResponseV6 != nil {
```

Comment:
also that structure of event leads to all this copies of the same code

Reply:
Agreed, will see how we can do this better.

Reply:
Done